Build error on fedora 30: clang-8: error: no such file or directory: 'external.googletest/lib/libgtest.a'

Trying to build the library like described in the README yields the following error for me:
```
FAILED: examples/libfuzzer/libfuzzer_example_test 
: && /usr/bin/clang++  -fno-exceptions -Werror -Wall -Wstring-conversion -g   examples/libfuzzer/CMakeFiles/libfuzzer_example_test.dir/libfuzzer_example_test.cc.o  -o examples/libfuzzer/libfuzzer_example_test  external.googletest/lib/libgtest.a external.googletest/lib/libgtest_main.a -pthread && :
clang-8: error: no such file or directory: 'external.googletest/lib/libgtest.a'
clang-8: error: no such file or directory: 'external.googletest/lib/libgtest_main.a'
[44/65] Performing configure step for 'external.expat'
```
I can find the respective files in the build directory, but not at the specified paths:

```
[vincent@vincent build]$ find . -name 'libgtest*.a'
./external.googletest/lib64/libgtest.a
./external.googletest/lib64/libgtest_main.a
./external.googletest/src/external.googletest-build/lib/libgtest.a
./external.googletest/src/external.googletest-build/lib/libgtest_main.a
```
Find my build process below. Anything I can do to fix this?

Thank you very much in advance.

I ran it like so:
```
[vincent@vincent build]$ LIB_PROTO_MUTATOR_DOWNLOAD_PROTOBUF=ON cmake .. -GNinja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Debug
-- The CXX compiler identification is Clang 8.0.0
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is Clang 8.0.0
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Could NOT find LibLZMA (missing: LIBLZMA_LIBRARY LIBLZMA_INCLUDE_DIR LIBLZMA_HAS_AUTO_DECODER LIBLZMA_HAS_EASY_ENCODER LIBLZMA_HAS_LZMA_PRESET) 
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.11") 
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_ADDRESS
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_ADDRESS - Success
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_SCOPE
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_SCOPE - Success
-- Performing Test LIB_PROTO_MUTATOR_HAS_NO_COVERAGE
-- Performing Test LIB_PROTO_MUTATOR_HAS_NO_COVERAGE - Success
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_FUZZER
-- Performing Test LIB_PROTO_MUTATOR_HAS_SANITIZE_FUZZER - Success
-- Performing Test LIB_PROTO_MUTATOR_HAS_NO_SANITIZE_FUZZER
-- Performing Test LIB_PROTO_MUTATOR_HAS_NO_SANITIZE_FUZZER - Success
-- Performing Test LIB_PROTO_MUTATOR_HAS_WSTRING_CONVERSION
-- Performing Test LIB_PROTO_MUTATOR_HAS_WSTRING_CONVERSION - Success
-- Found Protobuf: /usr/local/lib/libprotobuf.so;-pthread (found version "3.9.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vincent/Documents/Research/fuzzing/rust_protobuf_fuzzing/libprotobuf-mutator/build
```
And then:
```
[vincent@vincent build]$ ninja check
[18/65] Performing download step (git clone) for 'external.googletest'
Cloning into 'external.googletest'...
Note: checking out 'bf07131'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at bf07131c Merge pull request #2041 from ciband:chore/fix_library_json
[31/65] Performing download step (git clone) for 'external.expat'
Cloning into 'external.expat'...
Already on 'master'
Your branch is up to date with 'origin/master'.
[34/65] Performing configure step for 'external.googletest'
loading initial cache file /home/vincent/Documents/Research/fuzzing/rust_protobuf_fuzzing/libprotobuf-mutator/build/external.googletest/tmp/external.googletest-cache-Debug.cmake
-- The C compiler identification is Clang 8.0.0
-- The CXX compiler identification is Clang 8.0.0
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/clang++
-- Check for working CXX compiler: /usr/bin/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.16") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vincent/Documents/Research/fuzzing/rust_protobuf_fuzzing/libprotobuf-mutator/build/external.googletest/src/external.googletest-build
[35/65] Performing build step for 'external.googletest'
[1/8] Building CXX object googlemock/gtest/CMakeFiles/gtest_main.dir/src/gtest_main.cc.o
[2/8] Building CXX object googlemock/CMakeFiles/gmock_main.dir/src/gmock_main.cc.o
[3/8] Building CXX object googlemock/CMakeFiles/gmock.dir/src/gmock-all.cc.o
[4/8] Building CXX object googlemock/gtest/CMakeFiles/gtest.dir/src/gtest-all.cc.o
[5/8] Linking CXX static library lib/libgtest.a
[6/8] Linking CXX static library lib/libgtest_main.a
[7/8] Linking CXX static library lib/libgmock.a
[8/8] Linking CXX static library lib/libgmock_main.a
[36/65] Performing install step for 'external.googletest'
[0/1] Install the project...
[...]
[39/65] Linking CXX executable examples/libfuzzer/libfuzzer_example_test
FAILED: examples/libfuzzer/libfuzzer_example_test 
: && /usr/bin/clang++  -fno-exceptions -Werror -Wall -Wstring-conversion -g   examples/libfuzzer/CMakeFiles/libfuzzer_example_test.dir/libfuzzer_example_test.cc.o  -o examples/libfuzzer/libfuzzer_example_test  external.googletest/lib/libgtest.a external.googletest/lib/libgtest_main.a -pthread && :
clang-8: error: no such file or directory: 'external.googletest/lib/libgtest.a'
clang-8: error: no such file or directory: 'external.googletest/lib/libgtest_main.a'
[...]
[45/65] Performing download step (git clone) for 'external.libxml2'
Cloning into 'external.libxml2'...
warning: redirecting to https://gitlab.gnome.org/GNOME/libxml2.git/
Already on 'master'
Your branch is up to date with 'origin/master'.
ninja: build stopped: subcommand failed.

```

--- a/cmake/external/googletest.cmake
+++ b/cmake/external/googletest.cmake
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+include(GNUInstallDirs)
+
 set(GTEST_TARGET external.googletest)
 set(GTEST_INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/${GTEST_TARGET})
 
@@ -25,12 +27,12 @@ set(GTEST_BOTH_LIBRARIES ${GTEST_LIBRARIES} ${GTEST_MAIN_LIBRARIES})
 foreach(lib IN LISTS GTEST_BOTH_LIBRARIES)
   if (MSVC)
     if (CMAKE_BUILD_TYPE MATCHES Debug)
-      set(LIB_PATH ${GTEST_INSTALL_DIR}/lib/${lib}d.lib)
+      set(LIB_PATH ${GTEST_INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/${lib}d.lib)
     else()
-      set(LIB_PATH ${GTEST_INSTALL_DIR}/lib/${lib}.lib)
+      set(LIB_PATH ${GTEST_INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/${lib}.lib)
     endif()
   else()
-    set(LIB_PATH ${GTEST_INSTALL_DIR}/lib/lib${lib}.a)
+    set(LIB_PATH ${GTEST_INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/lib${lib}.a)
   endif()
   list(APPEND GTEST_BUILD_BYPRODUCTS ${LIB_PATH})
 

--- a/cmake/external/protobuf.cmake
+++ b/cmake/external/protobuf.cmake
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+include(GNUInstallDirs)
+
 # We only need protobuf_generate_cpp from FindProtobuf, and we are going to
 # override the rest with ExternalProject version.
 include (FindProtobuf)
@@ -31,9 +33,9 @@ ENDIF()
 
 foreach(lib ${PROTOBUF_LIBRARIES})
   if (MSVC)
-    set(LIB_PATH ${PROTOBUF_INSTALL_DIR}/lib/lib${lib}.lib)
+    set(LIB_PATH ${PROTOBUF_INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/lib${lib}.lib)
   else()
-    set(LIB_PATH ${PROTOBUF_INSTALL_DIR}/lib/lib${lib}.a)
+    set(LIB_PATH ${PROTOBUF_INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/lib${lib}.a)
   endif()
   list(APPEND PROTOBUF_BUILD_BYPRODUCTS ${LIB_PATH})
 

